packages/fx-core/src/plugins/solution/fx-solution/arm.ts

```diff
@@ -325,43 +327,28 @@ export async function getParameterJson(ctx: SolutionContext) {
     throw new Error("Failed to get target environment name from solution context.");
   }
 
-  let parameterFileName, parameterFolderPath, parameterTemplateFilePath;
-  if (isMultiEnvEnabled()) {
-    parameterFileName = parameterFileNameTemplateNew.replace("@envName", ctx.envInfo.envName);
-    parameterFolderPath = path.join(ctx.root, configsFolder);
-    parameterTemplateFilePath = path.join(
-      path.join(ctx.root, templateFolderNew),
-      parameterTemplateFileName
-    );
-  } else {
-    parameterFileName = parameterFileNameTemplate.replace("@envName", ctx.envInfo.envName);
-    parameterFolderPath = path.join(ctx.root, baseFolder, parameterFolder);
-    parameterTemplateFilePath = path.join(parameterFolderPath, parameterTemplateFileName);
-  }
-
+  const parameterFileName = parameterFileNameTemplate.replace(
+    "EnvNamePlaceholder",
+    ctx.envInfo.envName
+  );
+  const parameterFolderPath = path.join(ctx.root, configsFolder);
   const parameterFilePath = path.join(parameterFolderPath, parameterFileName);
-  let createNewParameterFile = false;
   try {
     await fs.stat(parameterFilePath);
   } catch (err) {
-    ctx.logProvider?.info(
-      `[${PluginDisplayName.Solution}] ${parameterFilePath} does not exist. Generate it using ${parameterTemplateFilePath}.`
+    ctx.logProvider?.error(`[${PluginDisplayName.Solution}] ${parameterFilePath} does not exist.`);
+    const returnError = new Error(
+      `[${PluginDisplayName.Solution}] ${parameterFilePath} does not exist.`
     );
-    createNewParameterFile = true;
+    throw returnUserError(returnError, "Solution", "ParameterFileNotExist");
   }
 
-  let parameterJson;
-  if (createNewParameterFile) {
-    await fs.ensureDir(parameterFolderPath);
-    if (isMultiEnvEnabled()) {
-      await fs.copyFile(parameterTemplateFilePath, parameterFilePath);
-    } else {
-      parameterJson = await getExpandedParameter(ctx, parameterTemplateFilePath, false); // do not expand secrets to avoid saving secrets to parameter file
-      await fs.writeFile(parameterFilePath, JSON.stringify(parameterJson, undefined, 2));
-    }
+  if (!isMultiEnvEnabled()) {
```

Comment:
I am curious why this check is needed. From my understanding, this PR has the assumption that bicep and multi-env feature flags are always enabled or disabled together. Other `isMultiEnvEnabled()` checks are all removed. Why is this one not removed?

Reply:
if multi-env not enable, parameter.default.json file will update the value, need to write back to the file;
once enable, azure.parameter.dev.json file no need to update the value. jsut copy

Reply:
sync up with @blackchoey  will remove the code.